### ISSUE TITLE
artifactory_permission_target is unable to refresh most of times with terraform-provider-artifactory_v8.2.2

### ISSUE THREAD

**Issue by nassah** [jfrog/terraform-provider-artifactory#760]
When terraform tries to refresh/plan `artifactory_permission_target` the error below is given [full trace error log](https://github.com/jfrog/terraform-provider-artifactory/files/11902611/terraform_plan.log).

The creation (apply) of the `artifactory_managed_user` and `artifactory_permission_target` is always successful, after creation of resources a plan (refresh) keeps crashing related to `artifactory_permission_target`.

To reproduce the issue from a snippet:
`terraform init`
`terraform apply` (successful)
`terraform plan` (FAILS) only way around this is disabling the refresh (`terraform plan -refresh=false`)

```
locals {
  managed_user_info = flatten([
    for system in ["sys0001","sys0002","sys0003","sys0004"] : [
      {
        user_principal_name        = "artifactory-write-${system}"
        disable_ui_access          = true
        internal_password_disabled = false
        admin                      = false
        profile_updatable          = false
        repository_permissions     = ["read", "write"]
      },
      {
        user_principal_name        = "artifactory-read-${system}"
        disable_ui_access          = true
        internal_password_disabled = false
        admin                      = false
        profile_updatable          = false
        repository_permissions     = ["read"]
      }
    ]
  ])
}

resource "artifactory_managed_user" "managed_user" {
  for_each = { for account in local.managed_user_info : account.user_principal_name => account }
  
  name                       = each.key
  email                      = "${each.value.user_principal_name}@mycorp.com"
  password                   = "Secret12#!"
  internal_password_disabled = each.value.internal_password_disabled
  disable_ui_access          = each.value.disable_ui_access
  admin                      = each.value.admin
  profile_updatable          = each.value.profile_updatable
  groups                     = []
}

resource "artifactory_remote_docker_repository" "remote_docker" {
  key                            = "dockerhub-remote"
  description                    = "Remote Dockerhub registry"
  external_dependencies_enabled  = true
  external_dependencies_patterns = ["**/*"]
  url                            = "https://registry-1.docker.io/"
  block_pushing_schema1          = true
  enable_token_authentication    = true
}

resource "artifactory_permission_target" "permission_target" {
  name = "provider-crash-permission-target"

  repo {
    repositories = ["dockerhub-remote"]
    actions {
      users {
        name = "artifactory-read-sys0001"
        permissions = [      
          "read",
        ]
      }
      users {
        name = "artifactory-write-sys0001"
        permissions = [
          "read",
          "write",
        ]                                                                                                                                   
      }
    }
  }

  depends_on = [
    artifactory_managed_user.managed_user
  ]
}

```

Artifactory (SaaS) version and revision:
version : 7.61.4
revision : 76104900

Terrform versions:
Terraform v1.5.1
provider registry.terraform.io/jfrog/artifactory v8.2.2

**Expected behavior**
Refreshing (plan) state and retrieving `artifactory_permission_target` without errors.

**Error output**

```
artifactory_managed_user.managed_user["artifactory-read-sys0002"]: Refreshing state... [id=artifactory-read-sys0002]
artifactory_remote_docker_repository.remote_docker: Refreshing state... [id=dockerhub-remote]
artifactory_permission_target.permission_target: Refreshing state... [id=provider-crash-permission-target]

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│
│   with artifactory_permission_target.permission_target,
│   on 00-main.tf line 47, in resource "artifactory_permission_target" "permission_target":
│   47: resource "artifactory_permission_target" "permission_target" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-artifactory_v8.2.2 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 184 [running]:
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySetElements({0x100ed3fa0, 0x14000c3a4b0}, {{{0x1400030d8c0, 0x7, 0xc}}, {0x100ed73c8, 0x14000c1b860}, {0x100ed73c8, 0x14000c2f680}}, 0x14000c3a450)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:145 +0xa3c
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySet({0x100ed3fa0, 0x14000c3a4b0}, {{{0x1400030d8c0, 0x7, 0xc}}, {0x100ed73c8, 0x14000c1b860}, {0x100ed73c8, 0x14000c2f680}}, 0x14000c3a450)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000c3a3f0}, {{{0x1400030d8c0, 0x7, 0xc}}, {0x100ed73c8, 0x14000c1b860}, {0x100ed73c8, 0x14000c2f680}}, 0x14000c3a450)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:79 +0x374
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualityObjectAttributes({0x100ed3fa0, 0x14000c3a3f0}, {{{0x1400079cd20, 0x6, 0xa}}, {0x100ed7338, 0x14000b481f8}, {0x100ed7338, 0x14000b48480}}, 0x14000c3a390)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_object.go:154 +0x6f0
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualityObject({0x100ed3fa0, 0x14000c3a3f0}, {{{0x1400079cd20, 0x6, 0xa}}, {0x100ed7338, 0x14000b481f8}, {0x100ed7338, 0x14000b48480}}, 0x14000c3a390)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_object.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000c3a360}, {{{0x1400079cd20, 0x6, 0xa}}, {0x100ed7338, 0x14000b481f8}, {0x100ed7338, 0x14000b48480}}, 0x14000c3a390)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:77 +0x32c
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySetElements({0x100ed3fa0, 0x14000c3a360}, {{{0x14000613600, 0x5, 0x8}}, {0x100ed73c8, 0x14000c1bad0}, {0x100ed73c8, 0x14000c2f8f0}}, 0x14000c3a300)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:152 +0x600
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySet({0x100ed3fa0, 0x14000c3a360}, {{{0x14000613600, 0x5, 0x8}}, {0x100ed73c8, 0x14000c1bad0}, {0x100ed73c8, 0x14000c2f8f0}}, 0x14000c3a300)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000c3a2a0}, {{{0x14000613600, 0x5, 0x8}}, {0x100ed73c8, 0x14000c1bad0}, {0x100ed73c8, 0x14000c2f8f0}}, 0x14000c3a300)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:79 +0x374
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualityObjectAttributes({0x100ed3fa0, 0x14000c3a2a0}, {{{0x140005e1bc0, 0x4, 0x6}}, {0x100ed7338, 0x14000b48288}, {0x100ed7338, 0x14000b48510}}, 0x14000c3a240)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_object.go:154 +0x6f0
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualityObject({0x100ed3fa0, 0x14000c3a2a0}, {{{0x140005e1bc0, 0x4, 0x6}}, {0x100ed7338, 0x14000b48288}, {0x100ed7338, 0x14000b48510}}, 0x14000c3a240)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_object.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000c3a1e0}, {{{0x140005e1bc0, 0x4, 0x6}}, {0x100ed7338, 0x14000b48288}, {0x100ed7338, 0x14000b48510}}, 0x14000c3a240)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:77 +0x32c
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySetElements({0x100ed3fa0, 0x14000c3a1e0}, {{{0x14000b2fec0, 0x3, 0x4}}, {0x100ed73c8, 0x14000c1bc80}, {0x100ed73c8, 0x14000c2faa0}}, 0x14000c3a180)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:152 +0x600
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySet({0x100ed3fa0, 0x14000c3a1e0}, {{{0x14000b2fec0, 0x3, 0x4}}, {0x100ed73c8, 0x14000c1bc80}, {0x100ed73c8, 0x14000c2faa0}}, 0x14000c3a180)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000c2fbf0}, {{{0x14000b2fec0, 0x3, 0x4}}, {0x100ed73c8, 0x14000c1bc80}, {0x100ed73c8, 0x14000c2faa0}}, 0x14000c3a180)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:79 +0x374
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualityObjectAttributes({0x100ed3fa0, 0x14000c2fbf0}, {{{0x1400054ada0, 0x2, 0x2}}, {0x100ed7338, 0x14000b482a0}, {0x100ed7338, 0x14000b48528}}, 0x14000c2fb90)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_object.go:154 +0x6f0
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualityObject({0x100ed3fa0, 0x14000c2fbf0}, {{{0x1400054ada0, 0x2, 0x2}}, {0x100ed7338, 0x14000b482a0}, {0x100ed7338, 0x14000b48528}}, 0x14000c2fb90)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_object.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000c2fb60}, {{{0x1400054ada0, 0x2, 0x2}}, {0x100ed7338, 0x14000b482a0}, {0x100ed7338, 0x14000b48528}}, 0x14000c2fb90)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:77 +0x32c
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySetElements({0x100ed3fa0, 0x14000c2fb60}, {{{0x14000c60410, 0x1, 0x1}}, {0x100ed73c8, 0x14000c1bce0}, {0x100ed73c8, 0x14000c2fb00}}, 0x140008fee58)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:152 +0x600
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEqualitySet({0x100ed3fa0, 0x14000c2fb60}, {{{0x14000c60410, 0x1, 0x1}}, {0x100ed73c8, 0x14000c1bce0}, {0x100ed73c8, 0x14000c2fb00}}, 0x140008fee58)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality_set.go:27 +0x314
github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata.ValueSemanticEquality({0x100ed3fa0, 0x14000d44030}, {{{0x14000c60410, 0x1, 0x1}}, {0x100ed73c8, 0x14000c1bce0}, {0x100ed73c8, 0x14000c2fb00}}, 0x140008fee58)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwschemadata/value_semantic_equality.go:79 +0x374
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.SchemaSemanticEquality({0x100ed3fa0, 0x14000d44030}, {{{0x100b5d826, 0x5}, {0x100ed9da8, 0x14000180410}, {{0x100ed89c0, 0x14000c05e60}, {0x100dc3660, 0x14000c03950}}}, ...}, ...)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwserver/schema_semantic_equality.go:126 +0x4f4
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x140002b0420, {0x100ed3fa0, 0x14000d44030}, 0x14000d44120, 0x140008ff3c0)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/fwserver/server_readresource.go:135 +0x714
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0x140002b0420, {0x100ed3fa0?, 0x14000927ec0?}, 0x14000d030c0)
	github.com/hashicorp/terraform-plugin-framework@v1.3.0/internal/proto5server/server_readresource.go:56 +0x1a4
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadResource({0x14000112960, 0x140001129f0, {0x1400028e820, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ReadResource.go:26 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14000149720, {0x100ed3fa0?, 0x14000926300?}, 0x14000eac6c0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:748 +0x3e8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x100e93f40?, 0x14000149720}, {0x100ed3fa0, 0x14000926300}, 0x140006bf8f0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:383 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140005b6000, {0x100ed8d00, 0x1400097eb60}, 0x14000d40b40, 0x14000898fc0, 0x101456eb0, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1345 +0xc64
google.golang.org/grpc.(*Server).handleStream(0x140005b6000, {0x100ed8d00, 0x1400097eb60}, 0x14000d40b40, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1722 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.54.0/server.go:966 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.54.0/server.go:964 +0x290

Error: The terraform-provider-artifactory_v8.2.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by alexhung** [comment]
@nassah This is most likely caused by #757 which is fixed in [v8.2.3](https://github.com/jfrog/terraform-provider-artifactory/releases/tag/v8.2.3) release yesterday.

**Comment by nassah** [comment]
@alexhung thank you for your reply, it is tested a few times and it seems that it has been fixed.
